quil-rs/src/program/graph.rs

```diff
@@ -516,22 +519,19 @@ fn terminate_working_block(
         return Ok(());
     }
 
-    let block = InstructionBlock::build(working_instructions.to_vec(), terminator, program)?;
+    // This leaves working_instructions and working_label in their respective "empty" states.
+    let block = InstructionBlock::build(std::mem::take(working_instructions), terminator, program)?;
     let label = working_label
-        .clone()
+        .take()
```

Comment:
`working_label` was previously set to `None` after processing. This avoids cloning and sets it to `None` at the same time.

This works because any error returned by this function is bubbled up past the point where invalid state could happen.

Reply:
That said, is it possible that the code could change in the future such that we try to recover from errors? If that were to happen, this assumption would be invalidated and it could lead to weird bugs.